bin/bastion.py

```diff
@@ -100,42 +91,16 @@ def create_help(header, options):
                         help = "Arguments to pass to the command")
 
     args = parser.parse_args()
+    bosslet_config = args.bosslet_config
 
-    if args.aws_credentials is None:
-        try:
-            print("AWS credentials not provided and AWS_CREDENTIALS is not defined, assuming IAM role")
-            session = aws.use_iam_role()
-        except Exception as e:
-            parser.print_usage()
-            print('Error: Could not assume IAM role due to:{}'.format(e))
-    elif args.aws_credentials is not None:
-        session = aws.create_session(args.aws_credentials)
-
-    if args.ssh_key is None:
-        parser.print_usage()
-        print("Error: SSH key not provided and SSH_KEY is not defined")
-        sys.exit(1)
-    if not os.path.exists(args.ssh_key):
-        parser.print_usage()
-        print("Error: SSH key '{}' does not exist".format(args.ssh_key))
-        sys.exit(1)
-
-    # This next step will make bastion work with 1.consul or 1.vault internal names.
-    boss_position = 1
-    try:
-        int(args.internal.split(".", 1)[0])
-        boss_position = 2
-    except ValueError:
-        pass
-
-    bastion_host = args.bastion if args.bastion else "bastion." + args.internal.split(".", boss_position)[boss_position]
-    bastion = aws.machine_lookup(session, bastion_host)
-    if args.private_ip:
-        private = args.internal
-    else:
-        private = aws.machine_lookup(session, args.internal, public_ip=False)
+    bastion = aws.machine_lookup(bosslet_config.session,
+                                 bosslet_config.names.bastion.dns) 
 
-    ssh = SSHConnection(args.ssh_key, (private, args.port, args.user), bastion)
+    ssh_target = SSHTarget(bosslet_config.ssh_key, args.ip, args.port, args.user)
+    bastions = [SSHTarget(bosslet_config.ssh_key, bastion, 22, 'ec2-user')]
```

Comment:
Do we enforce anywhere that the bastion EC2 instance should be? What if the user spins up a bastion with ubuntu, would the user 'ec2-user' still work? 
Not sure if I'm missing something?

Reply:
This is targeting the VPC's bastion machine, which is hardcoded (via `lib/constants.py` to be an Amazon AMI using the `ec2-user` account). I added a comment both to `lib/constants.py` for the VPC bastion AMI as well as to `bin/packer.py` for the Lambda Build Server AMI, which has the same assumptions made.